httpx/_models.py

```diff
@@ -838,7 +838,7 @@ def raise_for_status(self) -> None:
             message = message.format(self, error_type="Server Error")
             raise HTTPError(message, response=self)
 
-    def json(self, **kwargs: typing.Any) -> typing.Union[dict, list]:
+    def json(self, **kwargs: typing.Any) -> typing.Any:
```

Comment:
I'm surprised by:

> note that marking as : dict is not enough

Do you get the same error?

Reply:
I get a different, and confusing one:

```python
data: dict = httpx.get("http://example.com").json()
```

```console
error: Incompatible types in assignment (expression has type "Union[Dict[Any, Any], List[Any]]", variable has type "Dict[Any, Any]")
```

I assume it's because mypy doesn't allow just down-sizing a union like that - you'd need to make a proper `assert`, or `cast()`.

Reply:
Ugh, yeah, sounds like `Any` is the way to go here 👍